src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -120,15 +97,19 @@ class NativeCallStackStorage : public CHeapObj<mtNMT> {
   }
 
   NativeCallStackStorage(bool is_detailed_mode, int table_size = default_table_size)
-  : _arena(mtNMT), _table_size(table_size), _table(nullptr), _stacks(),
+  : _table_size(table_size), _table(nullptr), _stacks(),
     _is_detailed_mode(is_detailed_mode), _fake_stack() {
     if (_is_detailed_mode) {
-      _table = NEW_ARENA_ARRAY(&_arena, Link*, _table_size);
+      _table = NEW_C_HEAP_ARRAY(TableEntryIndex, _table_size, mtNMT);
       for (int i = 0; i < _table_size; i++) {
-        _table[i] = nullptr;
+        _table[i] = TableEntryStorage::nil;
       }
     }
   }
```

Comment:
Could you move this, and the dtor, over to the cpp file, too? You probably can remove the allocation.hpp include then. Less include deps are always good.